pkg/bgpv1/types/conversions.go

```diff
@@ -265,3 +266,207 @@ func ToAgentFamily(fam v2alpha1.CiliumBGPFamily) Family {
 		Safi: ParseSafi(fam.Safi),
 	}
 }
+
+// ToNeighborV1 converts a CiliumBGPNeighbor to Neighbor which can be used
+// with Router API. The caller must ensure that the an is not nil.
+func ToNeighborV1(an *v2alpha1.CiliumBGPNeighbor, password string) *Neighbor {
```

Comment:
While we should move this to `reconciler` package, we will keep it here for the moment, because some tests in the `gobgp` package are still relying on CRD and we need to use conversion function from there. Importing `reconciler` package from `gobgp` package causes cyclic import.